pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2280,3 +2312,79 @@ func TestUpdateNodePorts(t *testing.T) {
 		}
 	}
 }
+
+func TestServiceRegistryCreateWithNetworkOfExternalIPAM(t *testing.T) {
+	storage, registry, server := NewTestREST(t, nil)
+	storage.networks = &dummyNetworkGetter{ipamType: "External"}
+	defer server.Terminate(t)
+
+	svc := &api.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Tenant:    "system",
+			Namespace: "default",
+		},
+		Spec: api.ServiceSpec{
+			Selector:        map[string]string{"bar": "baz"},
+			SessionAffinity: api.ServiceAffinityNone,
+			Type:            api.ServiceTypeClusterIP,
+			Ports: []api.ServicePort{{
+				Port:       6502,
+				Protocol:   api.ProtocolTCP,
+				TargetPort: intstr.FromInt(6502),
+			}},
+		},
+	}
+	ctx := genericapirequest.NewDefaultContext()
+	created_svc, err := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	created_service := created_svc.(*api.Service)
+
+	if created_service.Spec.ClusterIP != "" {
+		t.Errorf("expected no ClusterIP on creation; got %s", created_service.Spec.ClusterIP)
+	}
+
+	srv, err := registry.GetService(ctx, svc.Name, &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if srv == nil {
+		t.Errorf("Failed to find service: %s", svc.Name)
+	}
+
+	if srv.Spec.ClusterIP != "" {
+		t.Fatalf("expected no ClusterIP on query; got %s", srv.Spec.ClusterIP)
+	}
+}
+
+func TestServiceRegistryCreateWithoutExistingNetwork(t *testing.T) {
+	storage, _, server := NewTestREST(t, nil)
+	storage.networks = &dummyNetworkGetter{err: fmt.Errorf("network not found")}
+	defer server.Terminate(t)
+
+	svc := &api.Service{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Add tenant and namespace in ObjectMeta.

Reply:
will add tenant + namespace as suggested